### ISSUE TITLE
Give Options for admin user in prompts when attempting to Approve a Pending Reservation. Also commented here - limitations of SO user type when submit a res.

### ISSUE THREAD

**Issue by ralch34** [philrhoy/IMCCVenueReservationSystem2023#34]
GIVE OPTIONS FOR ADMIN USER IN PROMPTS when attempting to Approve a Pending Reservation.

Sulod sa "update" button then ang DSA mu-attempt click on "approve" then Apply Changes.

System must prompt admin user... 

****
Example no. 1
"Are you sure you want to Approve this Reservation?, no conflict detected, no missing uploads detected."
"Yes/Proceed" and "No" buttons..
______
Example no. 2
"Warnings: There is no Uploaded Letter of Approval in this reservation, missing uploads detected!"
"Are you sure you want to Approve this Reservation? or Upload your Letter of Approval?"
"Yes/Proceed" and "No" buttons..

This is the case naay times considered ra walay Letter of Approval, as long as naa ang uploaded Student Activity Form.
Magprovide gihapon tag Prompt... gihapon...

_____
Example no. 3
"Warnings: There is no Uploaded Student Activity Form in this reservation., missing uploads detected!"
"You cannot proceed unless upload the supporting documents for this reservation."
"Back to Edit" nga button.

in this case, possible ba ma remove by the DSA ang uploaded files in an STO Reservation Submission? if yess you can add this prompt...

_____
Example no 4.
"Warnings: You are attempting to approve this Reservation that conflicts with already approved reservation RES06266, Conflict detected!"
(System Displays RES06266 Venue and Sched Details)
"You cannot proceed unless assign this reservation to a different venue, date, or time range"
"Back to Edit" nga button.

in this case, possible ma-accidentally change sa DSA ang chosen venue, date, or time range of the opened Pending Reservations...

**Comment by philrhoy** [comment]
Can I ask the limitation sa SO? regarding sa pag submit ug reservation

**Comment by ralch34** [comment]
> Can I ask the limitation sa SO? regarding sa pag submit ug reservation

Here's, STO limitations.
1. Can upload image file format and pdf on "Upload Fully-Signed S.A.F" & "Upload Letter of Approval"
2. STO users CAN submit reservations with missing "upload letter of approval". In other words, All textboxes, from activity titles to selected date and time and venue are mandatory entries, but if STO user submits with a missing upload "letter of approval" - pop-up prompt or warning is needed for STO.
   
_2. (part 2 not related to STO ) for DSA user side, since DSA user can "create reservation", still apply this limitation. Some circumstances the DSA proposes student activities without a letter of approval. It's weird because usually, DSA entries for Reservation are walk-ins from outside organizations who like to have an event in the school, they usually have a letter of approval. The inside activities of the school (let's say from other departments) they sometimes delay or fail to present their "letter of approval" to the DSA. However, if "Student Activity Form" is Fully signed and presented to the Head of DSA, the head approves it._

3. STOs only see their own reservation submissions in the "reservations" tab. 
4. If reservation status is still pending, STO user can use "Update" or "edit" button.
5. IF Res with Approved Status will lock the Update buttons for STO users.
Basically once approved, only the DSA and PTC can update the reservations due to the Reservation Notes (PTC) and of course the maintain integrity of such approved reservation (DSA)- if Head of DSA sees the uploaded files are valid, then after approving it locks "update" for STO, so that no unauthorized changes are made by the STO user. 
6. Reservations with Rejected Status, STO users can still "Update" the reservations.
7. So far wala pay "delete" reservation, so I have no idea unsay limitations for STO.

**Comment by ralch34** [comment]
IN DSA (CREATE RESERVATION)

1. Can upload image file format and pdf on "Upload Fully-Signed S.A.F" & "Upload Letter of Approval"
DSA users CAN submit reservations with missing "upload letter of approval". In other words, All textboxes, from activity titles to selected date and time and venue (INCLUDING the FULLY-SIGNED Student Activity form) are mandatory entries. The letter of approval is optional. DSA user submits with a missing upload "letter of approval" - pop-up prompt or warning is needed for STO.
2. (part 2 not related to STO ) for DSA user side, since DSA user can "create reservation", still apply this limitation. Some circumstances the DSA proposes student activities without a letter of approval. It's weird because usually, DSA entries for Reservation are walk-ins from outside organizations who like to have an event in the school, they usually have a letter of approval. The inside activities of the school (let's say from other departments) they sometimes delay or fail to present their "letter of approval" to the DSA. However, if "Student Activity Form" is Fully signed and presented to the Head of DSA, the head approves it.

3. DSA user can see their own reservation submissions in the "reservations" tab if using filter "User type". 
4. FOR DSA cannot directly APPROVE when create reservation, it has to go through Pending status. This is to give the DSA user a chance or time to decide whether he commits approval of the created reservation or not. So pending reservaitons for DSA user can use "Update" or "edit" button as usual.
5. If Res is Approved, DSA can still Update since we have an "update record only" option. 
6. Reservations with Rejected Status, DSA users can still "Update" the reservations into Approved.. or update record only. Don't forget the "notes" are editable too.
7. So far wala pay "delete" reservation, so I have no idea unsay limitations for DSA.

**Comment by ralch34** [comment]
@philrhoy 
as i checked goods na ni.... ( also checked sa approve option and update record only option)

![Screenshot_45](https://github.com/philrhoy/IMCCVenueReservationSystem2023/assets/90671718/a3b1b416-348e-48b3-ae6d-35868e27f495)

but.... it will be cooler if displays additional prompts Like schedule conflict (if detected - the DSA still has ability to UPDATE the start date, end date, start tine, .... and son ) as suggested from the original post above.

@philrhoy, I like you to open up to me... of much work it needs to be done in terms of working for the "conflict detection" in schedules. 

"
Example no. 1
"Are you sure you want to Approve this Reservation?, **no schedule conflict detected,**"
"Yes/Proceed" and "No" buttons..

and so on...."   

Priority lang ang "Schedule Conflict Detection" nga part

**Comment by ralch34** [comment]
In cases of Rejected Reservations given a 2nd chance (after edits) if DSA changes its status to "approved".

The reservation MAY displayed again if ever it does not conflict with other already approved reservation schedules.

This leads us to another issue, "When attempting to create two reservations with the same venue, date, and time range, an error code pops up. - a suggestion that a Conflict detection is needed." details will be mentioned in this issue later...

Let's go back on topic, to avoid other issues that may arise, you put a WARNING PROMPT FOR DSA Admin user when attempting to change a Rejected Reservation into an Approved Reservation.

After or before?? apply changes to the status (proceed button).

a warning box appears. Example no. 2
"Warnings: You are attempting to approve this Rejected Reservation schedule that conflicts with already approved reservation RES002823, Conflict detected!"
(System Displays RES002823 Venue and Sched Details)
"You cannot proceed unless assign this reservation to a different venue, date, or time range"
"Back to Edit" nga button.

**Comment by ralch34** [comment]
> In cases of Rejected Reservations given a 2nd chance (after edits) if DSA changes its status to "approved".
> 
> The reservation MAY displayed again if ever it does not conflict with other already approved reservation schedules.
> 
> This leads us to another issue, "When attempting to create two reservations with the same venue, date, and time range, an error code pops up. - a suggestion that a Conflict detection is needed." details will be mentioned in this issue later...
> 
> Let's go back on topic, to avoid other issues that may arise, you put a WARNING PROMPT FOR DSA Admin user when attempting to change a Rejected Reservation into an Approved Reservation.
> 
> After or before?? apply changes to the status (proceed button).
> 
> a warning box appears. Example no. 2 "Warnings: You are attempting to approve this Rejected Reservation schedule that conflicts with already approved reservation RES002823, Conflict detected!" (System Displays RES002823 Venue and Sched Details) "You cannot proceed unless assign this reservation to a different venue, date, or time range" "Back to Edit" nga button.

this is connected to issue #33
